src/venvstacks/stacks.py

```diff
@@ -1527,97 +1546,124 @@ def env_spec(self) -> ApplicationSpec:
+        for env in self._linked_environments():
+            dynlib_path = env.dynlib_path
+            if dynlib_path is not None:
+                yield env.get_deployed_path(dynlib_path)
+
     def link_layered_environments(
         self, runtime: RuntimeEnv, frameworks: Mapping[LayerBaseName, FrameworkEnv]
     ) -> None:
-        self.link_base_runtime_paths(runtime)
+        self.link_base_runtime(runtime)
         constraints_paths = self.linked_constraints_paths
         if not constraints_paths:
             raise BuildEnvError("Failed to add base environment constraints path")
         # The runtime site-packages folder is added here rather than via pyvenv.cfg
         # to ensure it appears in sys.path after the framework site-packages folders
-        pylib_paths = self.linked_pylib_paths
-        dynlib_paths = self.linked_dynlib_paths
         fw_envs = self.linked_frameworks
-        if pylib_paths or dynlib_paths or fw_envs:
+        if fw_envs:
             raise BuildEnvError("Layered application environment already linked")
-        for env_spec in self.env_spec.frameworks:
-            env = frameworks[env_spec.name]
-            fw_envs.append(env)
-            constraints_paths.append(env.requirements_path)
-            install_target_path = Path(env.install_target)
-
-            def _fw_env_path(build_path: Path) -> Path:
-                relative_path = build_path.relative_to(env.env_path)
-                return install_target_path / relative_path
-
-            pylib_paths.append(_fw_env_path(env.pylib_path))
-            if env.dynlib_path is not None:
-                dynlib_paths.append(_fw_env_path(env.pylib_path))
-        runtime_target_path = Path(runtime.install_target)
-
-        def _runtime_path(build_path: Path) -> Path:
-            relative_path = build_path.relative_to(runtime.env_path)
-            return runtime_target_path / relative_path
-
-        pylib_paths.append(_runtime_path(runtime.pylib_path))
-        if runtime.dynlib_path is not None:
-            dynlib_paths.append(_runtime_path(runtime.dynlib_path))
-
-    def _link_layered_environment(self) -> None:
-        # Create sitecustomize file
-        sc_dir_path = self.pylib_path
+        for fw_env_spec in self.env_spec.frameworks:
+            fw_env = frameworks[fw_env_spec.name]
+            fw_envs.append(fw_env)
+            constraints_paths.append(fw_env.requirements_path)
+
+    @staticmethod
+    def _render_sitecustomize(
+        relative_prefix: Path,
+        pylib_paths: Iterable[Path],
+        dynlib_paths: Iterable[Path],
+    ) -> str:
         sc_contents = [
             "# Automatically generated by venvstacks",
             "import site",
             "import os",
             "from os.path import abspath, dirname, join as joinpath",
-            "# Allow loading modules and packages from framework environments",
+            "# Allow loading modules and packages from linked environments",
```

Comment:
This text change is what changed the hashes for even the unversioned application layers without any versioned dependencies.